src/OpenTelemetry.Api/Context/Propagation/TextMapPropagator.cs

```diff
@@ -259,7 +259,7 @@ private static byte HexCharToByte(char c)
                 return Convert.ToByte(c);
             }
 
-            throw new ArgumentOutOfRangeException(nameof(c), $"Invalid character: {c}.");
+            throw new ArgumentOutOfRangeException(nameof(c), c, $"Invalid character: {c}.");
```

Comment:
A general question - do we want all the exception messages to be ended with period `.` or not.
I don't have strong opinion on either way.
I would suggest that we 1) follow the .NET convention (if there is one) and 2) make it consistent in this repo.

Reply:
There is no .NET convention. Looking a microsoft AspNetCore repo, some places they have "." others they just throw stuff like: `throw new ArgumentNullException(nameof(hostBuilder));`
https://github.com/dotnet/aspnetcore/blob/master/src/Azure/AzureAppServicesIntegration/src/AppServicesWebHostBuilderExtensions.cs

So guess it's pretty much up to the team / developers to figure it out.

Reply:
Don't know if it is out of scope for this issue? And perhaps a talk in the team what is best for Opentelemetry

Reply:
Yeah, no need to block this PR.

Reply:
https://tenor.com/view/git-merge-gitmerge-gif-18010083